### ISSUE TITLE
Kernel error caused by a decorator

### ISSUE THREAD

**Issue by Tuyixiang** [jupyter/notebook#5931]
## Bug Description
Just installed notebook with `pip`. Each time I try to open a notebook, a "Kernel Error" will happen. 

As I look into the source code, there's a problem with `jupyter-client` using `traitlets`, which I manage to circumvent by making a change in `jupyter-client`. 

## Desktop
- OS: Windows 10
- Browser: Microsoft Edge
- Version: 
  - Python 3.9.1
  - `jupyter-client 6.1.8`
  - `traitlets 5.0.5`

## Crash Info
![image](https://user-images.githubusercontent.com/34904872/104006299-4316ef00-51e1-11eb-947f-d8092b1df89d.png)
```
[E 18:28:19.796 NotebookApp] Uncaught exception POST /api/sessions (::1)
    HTTPServerRequest(protocol='http', host='localhost:8888', method='POST', uri='/api/sessions', version='HTTP/1.1', remote_ip='::1')
    Traceback (most recent call last):
      File "......\python39\lib\site-packages\tornado\web.py", line 1704, in _execute
        result = await result
      File "......\python39\lib\site-packages\tornado\gen.py", line 769, in run
        yielded = self.gen.throw(*exc_info)  # type: ignore
      File "......\python39\lib\site-packages\notebook\services\sessions\handlers.py", line 69, in post
        model = yield maybe_future(
      File "......\python39\lib\site-packages\tornado\gen.py", line 762, in run
        value = future.result()
      File "......\python39\lib\site-packages\tornado\gen.py", line 769, in run
        yielded = self.gen.throw(*exc_info)  # type: ignore
      File "......\python39\lib\site-packages\notebook\services\sessions\sessionmanager.py", line 88, in create_session
        kernel_id = yield self.start_kernel_for_session(session_id, path, name, type, kernel_name)
      File "......\python39\lib\site-packages\tornado\gen.py", line 762, in run
        value = future.result()
      File "......\python39\lib\site-packages\tornado\gen.py", line 769, in run
        yielded = self.gen.throw(*exc_info)  # type: ignore
      File "......\python39\lib\site-packages\notebook\services\sessions\sessionmanager.py", line 100, in start_kernel_for_session
        kernel_id = yield maybe_future(
      File "......\python39\lib\site-packages\tornado\gen.py", line 762, in run
        value = future.result()
      File "......\python39\lib\site-packages\notebook\services\kernels\kernelmanager.py", line 176, in start_kernel
        kernel_id = await maybe_future(self.pinned_superclass.start_kernel(self, **kwargs))
      File "......\python39\lib\site-packages\jupyter_client\multikernelmanager.py", line 184, in start_kernel
        km, kernel_name, kernel_id = self.pre_start_kernel(kernel_name, kwargs)
      File "......\python39\lib\site-packages\jupyter_client\multikernelmanager.py", line 169, in pre_start_kernel
        km = self.kernel_manager_factory(connection_file=os.path.join(
      File "......\python39\lib\site-packages\jupyter_client\multikernelmanager.py", line 83, in create_kernel_manager
        km = kernel_manager_ctor(*args, **kwargs)
      File "......\python39\lib\site-packages\traitlets\config\configurable.py", line 82, in __init__
        super(Configurable, self).__init__(**kwargs)
      File "......\python39\lib\site-packages\traitlets\traitlets.py", line 1080, in __init__
        super_kwargs[key] = value
      File "......\python39\lib\contextlib.py", line 124, in __exit__
        next(self.gen)
      File "......\python39\lib\site-packages\traitlets\traitlets.py", line 1214, in hold_trait_notifications
        self.notify_change(change)
      File "......\python39\lib\site-packages\traitlets\traitlets.py", line 1227, in notify_change
        return self._notify_observers(change)
      File "......\python39\lib\site-packages\traitlets\traitlets.py", line 1264, in _notify_observers
        c(event)
      File "......\python39\lib\site-packages\traitlets\traitlets.py", line 888, in compatible_observer
        return func(self, change)
    TypeError: _kernel_spec_manager_changed() takes 1 positional argument but 2 were given
[W 18:28:19.801 NotebookApp] Unhandled error
```

## Problem
In `jupyter-client`, `manager.py` line 66:
```python
@observe('kernel_spec_manager')
@observe_compat
def _kernel_spec_manager_changed(self):
    self._kernel_spec = None
```
As is implemented in `traitlets`, `traitlets.py` line 861, `observe_compat` passes 2 arguments to the decorated function:
```python
def observe_compat(func):
    def compatible_observer(self, change_or_name, old=Undefined, new=Undefined):
        # code omitted
        return func(self, change) # <- here
    return compatible_observer
```
Adding another argument to `_kernel_spec_manager_changed` seems to solve the problem:
```python
@observe('kernel_spec_manager')
@observe_compat
def _kernel_spec_manager_changed(self, _change):
    self._kernel_spec = None
```

**Comment by jonsnowseven** [comment]
👍

**Comment by tom5079** [comment]
Seems like it's fixed on jupyter-client version 6.1.10
upgrade it by `pip install jupyter-client==6.1.10`

**Comment by Tuyixiang** [comment]
> Seems like it's fixed on jupyter-client version 6.1.10
> upgrade it by `pip install jupyter-client==6.1.10`

Yep it is fixed.
Just wonder why I only saw 6.1.8 in the first place.
